pymc4/distributions/half_cauchy.py

```diff
@@ -0,0 +1,160 @@
+
+  Examples of initialization of one or a batch of distributions.
+
+  ```python
+  # Define a single scalar HalfNormal distribution.
+  dist = tf.contrib.distributions.HalfNormal(scale=3.0)
+
+  # Evaluate the cdf at 1, returning a scalar.
+  dist.cdf(1.)
+
+  # Define a batch of two scalar valued HalfNormals.
+  # The first has scale 11.0, the second 22.0
+  dist = tf.contrib.distributions.HalfNormal(scale=[11.0, 22.0])
+
+  # Evaluate the pdf of the first distribution on 1.0, and the second on 1.5,
+  # returning a length two tensor.
+  dist.prob([1.0, 1.5])
+
+  # Get 3 samples, returning a 3 x 2 tensor.
+  dist.sample([3])
+  ```
+
+  """
+
+  def __init__(self,
+               scale,
+               validate_args=False,
+               allow_nan_stats=True,
+               name="HalfCauchy"):
+    """Construct HalfNormals with scale `scale`.
+
+    Args:
+      scale: Floating point tensor; the scales of the distribution(s).
+        Must contain only positive values.
+      validate_args: Python `bool`, default `False`. When `True` distribution
+        parameters are checked for validity despite possibly degrading runtime
+        performance. When `False` invalid inputs may silently render incorrect
+        outputs.
+      allow_nan_stats: Python `bool`, default `True`. When `True`,
+        statistics (e.g., mean, mode, variance) use the value "`NaN`" to
+        indicate the result is undefined. When `False`, an exception is raised
+        if one or more of the statistic's batch members are undefined.
+      name: Python `str` name prefixed to Ops created by this class.
+    """
+    parameters = dict(locals())
+    with tf.name_scope(name, values=[scale]) as name:
+      with tf.control_dependencies([tf.assert_positive(scale)]
+                                   if validate_args else []):
+        self._scale = tf.identity(scale, name="scale")
+    super(HalfNormal, self).__init__(
+        dtype=self._scale.dtype,
+        reparameterization_type=tf.distributions.FULLY_REPARAMETERIZED,
+        validate_args=validate_args,
+        allow_nan_stats=allow_nan_stats,
+        parameters=parameters,
+        graph_parents=[self._scale],
+        name=name)
+
+  @staticmethod
+  def _param_shapes(sample_shape):
+    return {"scale": tf.convert_to_tensor(sample_shape, dtype=tf.int32)}
+
+  @property
+  def scale(self):
+    """Distribution parameter for the scale."""
+    return self._scale
+
+  def _batch_shape_tensor(self):
+    return tf.shape(self.scale)
+
+  def _batch_shape(self):
+    return self.scale.shape
+
+  def _event_shape_tensor(self):
+    return tf.constant([], dtype=tf.int32)
+
+  def _event_shape(self):
+    return tensor_shape.scalar()
+
+  def _sample_n(self, n, seed=None):
+    shape = tf.concat([[n], self.batch_shape_tensor()], 0)
+    probs = tf.random_uniform(
+        shape=shape, minval=0., maxval=1., dtype=self.dtype, seed=seed)
+    return tf.abs(sampled * self.scale)
+
+  def _prob(self, x):
```

Comment:
Don't we get all of that from the bijection applied to the Cauchy distribution?

Reply:
As @jvdillon mentioned in the tensorflow_probability [issue](https://github.com/tensorflow/probability/issues/100#issuecomment-405099938), the `AbsoluteValue` solution resembles HalfCauchy with similar properties but isn't exactly that. 

From what I understand, `AbsoluteValue` gives a modified scaled up version of Cauchy resulting in similar properties to HalfCauchy(with same scale as Cauchy described earlier) only on the positive side. This is demonstrated in this [notebook](https://gist.github.com/sharanry/e9552d3a74ec49e32e1e7fea4739e2b7). 

PS: The notebook uses HalfCauchy defined using absolute value.

IMO Only way to have a proper HalfCauchy distribution is to define it from scratch like tensorflow_probability does with [other distributions](https://github.com/tensorflow/probability/tree/master/tensorflow_probability/python/distributions).

Reply:
I see, that makes sense.